File: hw_1403/src/main/java/com/example/hw_1403/recyclerview/ItemTouchCallback.java
package com.example.hw_1403.recyclerview;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.helper.ItemTouchHelper;

public class ItemTouchCallback extends ItemTouchHelper.SimpleCallback {

    private final StudentsAdapter adapter;

    public ItemTouchCallback(final RecyclerView recycler, final StudentsAdapter adapter) {
        super(ItemTouchHelper.UP | ItemTouchHelper.DOWN, ItemTouchHelper.START
                | ItemTouchHelper.END);
        this.adapter = adapter;
    }

    @Override
    public boolean onMove(@NonNull final RecyclerView recyclerView,
                          @NonNull final RecyclerView.ViewHolder viewHolder,
                          @NonNull final RecyclerView.ViewHolder viewHolder1) {
        adapter.onItemMove(viewHolder.getAdapterPosition(), viewHolder1.getAdapterPosition());

        return true;
    }

    @Override
    public void onSwiped(@NonNull final RecyclerView.ViewHolder viewHolder, final int i) {
        final int adapterPosition = viewHolder.getAdapterPosition();

        if (RecyclerView.NO_POSITION != adapterPosition) {
            adapter.onItemDismiss(adapterPosition);
        }
    }
}

File: app/src/main/java/com/shvants/runninglife/utils/ItemTouchCallback.kt
package com.shvants.runninglife.utils

import androidx.annotation.NonNull
import androidx.recyclerview.widget.ItemTouchHelper
import androidx.recyclerview.widget.RecyclerView

class ItemTouchCallback(recycler: RecyclerView,
                        private val adapter: IAdapter) :
        ItemTouchHelper.SimpleCallback(
                ItemTouchHelper.UP or ItemTouchHelper.DOWN,
                ItemTouchHelper.START or ItemTouchHelper.END) {

    override fun onMove(@NonNull recyclerView: RecyclerView,
                        @NonNull viewHolder: RecyclerView.ViewHolder,
                        @NonNull viewHolder1: RecyclerView.ViewHolder): Boolean {

//        adapter.onItemMove(viewHolder.adapterPosition, viewHolder1.adapterPosition)

        return true
    }

    override fun onSwiped(@NonNull viewHolder: RecyclerView.ViewHolder, direction: Int) {}
}

